eureka2-client/src/main/java/com/netflix/eureka2/client/resolver/ServerResolvers.java

```diff
@@ -43,16 +47,18 @@ public static ServerResolver forDnsName(String dnsName, int port) {
     }
 
     public static ServerResolver fromWriteServer(ServerResolver writeServerResolver, String readClusterVip) {
-        return fromWriteServer(writeServerResolver, readClusterVip, EurekaClientMetricFactory.clientMetrics());
+        return fromWriteServer(writeServerResolver, readClusterVip, clientMetrics(), registryMetrics());
     }
 
     public static ServerResolver fromWriteServer(ServerResolver writeServerResolver,
                                                  String readClusterVip,
-                                                 EurekaClientMetricFactory metricFactory) {
+                                                 EurekaClientMetricFactory clientMetricFactory,
+                                                 EurekaRegistryMetricFactory registryMetricFactory) {
         return new EurekaServerResolverBuilder()
                 .withBootstrapResolver(writeServerResolver)
                 .withReadServerInterest(Interests.forVips(readClusterVip))
-                .withMetricFactory(metricFactory)
+                .withMetricFactory(clientMetricFactory)
```

Comment:
the withMetricFactory overload seems confusing, we can be more specific with the type of metric factories?